### PR TITLE
FO-968: Increment Nonce for CheckTx run mode without executing EVM transaction 

### PR DESCRIPTION
**Pull request by kevinssgh**
1. Modified ValidateUnsigned trait - removed redundant validate call in pre execute function
2. Modified evm.py to use pending nonce when broadcasting a send transaction
3. Call apply for transaction check but only run pre-execute function
4. Updated startnodes.sh to use the variable $BIN instead of hard-coding release path

### CODE REVIEW

File: tools/devnet/startnodes.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ if [ -z "$Node" ] || ([ ! -z "$Node" ] && [ "$Node" = "$node" ]); then
         ENABLE_LEDGER_SERVICE=true \
         ENABLE_ETH_API_SERVICE=true \
         ENABLE_QUERY_SERVICE=true \
-        release/bin/abcid $DEVNET/$node >> $DEVNET/$node/abcid.log 2>&1  &
+        $BIN/abcid $DEVNET/$node >> $DEVNET/$node/abcid.log 2>&1  &
```
Comment (ws4charlie) [inline tools/devnet/startnodes.sh:24]: I remember that $BIN already points to release/bin by default in env.sh.

**Review by ws4charlie — COMMENTED** [review]
The change looks good. CI failed.

File: tools/devnet/startnodes.sh (hunk 1 above)
Comment (kevinssgh) [reply tools/devnet/startnodes.sh:24]: Yes, by default it is release but if I want to change it to debug, I should only have to change it in one place

Comment (ws4charlie) [reply tools/devnet/startnodes.sh:24]: looks great after modification.

**Review by simonjiao — COMMENTED** [review]
Looks good to me.  But will it be better if we just remove other operations(such as PendingTransactions, events, transaction index etc) but still executing evm transaction. Executing a transaction, after all,  is more complex than increasing account nonce.

**Comment by kevinssgh** [comment]
> Looks good to me. But will it be better if we just remove other operations(such as PendingTransactions, events, transaction index etc) but still executing evm transaction. Executing a transaction, after all, is more complex than increasing account nonce.

Do you mean creating a separate function "Pre-execute" in the "Applyable" trait for CheckedTransaction where we put the other operations? I did consider that but the "apply" function requires some data to be passed from the pre-execute to the post execute function. Also doing this would complicate the implementation at the modules layer and I was trying to minimize the impact of the change.

Also the main reason we were doing this was to increase the pending nonce without running a simulation of the transaction. The main reason for that was to deliver all the transactions including reverted ones in order to get a receipt. 

If I didn't completely understand, we can set up a call with Charlie.

**Comment by ws4charlie** [comment]
I think the key point is that by skipping executing the whole transaction, we'll have a consistent behaviour with other Ethereum chains. It's more about the compatibility than the correctness.

**Review by ws4charlie — APPROVED** [review]
Looks good

**Pull request merged**